via-router/src/router.rs

```diff
@@ -271,13 +271,11 @@ impl<'a, 'b, T> Iterator for Traverse<'a, 'b, T> {
         let branches = &mut self.branches;
 
         loop {
-            if let Some(binding) = depth {
-                if let Some(next) = binding.next() {
-                    return Some(next);
-                }
+            if let Some(next) = depth.as_mut().and_then(Iterator::next) {
```

Comment:
Performance doesn't seem to change here. The Rust compiler is doing a good job at inlining this to a single pattern match.